File: Chapter4/summarymapper.rb
#!/usr/bin/env ruby

while line = gets
	puts "total\t1"
	parts = line.split("\t")
	puts "badline\t1" if parts.size != 6
	puts "sighted\t1" if !parts[0].nil?
	puts "recorded\t1" if !parts[1].nil?
	puts "location\t1" if !parts[2].nil?
	puts "shape\t1" if !parts[3].nil?
	puts "duration\t1" if !parts[4].nil?
	puts "description\t1" if !parts[5].nil?
end

File: Chapter4/summarymapper.py
#!/usr/bin/env python

import sys

for line in sys.stdin:
	words = line.split("\t")
	keys = ["sighted","recorded","location","shape","duration","description"]
	sys.stdout.write("total\t1\n")
	if(len(words) != 6):
		sys.stdout.write("badline\t1\n")
	for i in range(6 if len(words) >= 6 else 4):
		sys.stdout.write(keys[i] + "\t1\n")


